File: src/services/Models/Character.ts
export class Character {
  public id: string
  public name: string
  public knowCharacterIds: string[]

  constructor(id: string, name: string, knowCharacterIds: string[] = []) {
    this.id = id
    this.name = name
    this.knowCharacterIds = knowCharacterIds
  }
}

File: src/LoreWeave.Domain/Entities/Characters/Character.cs
using LoreWeave.Domain.Entities.Characters.Commands;

namespace LoreWeave.Domain.Entities.Characters;

public sealed class Character
{
    public Character(CreateCharacter createCharacter)
    {
        Id = createCharacter.Id;
        Name = createCharacter.Name;
        Version = 1;
    }

    public Character(CreateCharacter createCharacter, int version) : this(createCharacter) => Version = version;

    public Ulid Id { get; private init; }

    public string Name { get; private set; }

    public int Version { get; private set; }
}
